File: library/banner/src/main/java/com/example/banner/view/ViewPagerScroller.java
package com.example.banner.view;

import android.content.Context;
import android.widget.Scroller;

public class ViewPagerScroller extends Scroller {

    // 滑动速度,值越大滑动越慢，滑动太快会使3d效果不明显
    private int scrollDuration = 800;

    private boolean zero;

    public ViewPagerScroller(Context context) {
        super(context);
    }

    @Override
    public void startScroll(int startX, int startY, int dx, int dy, int duration) {
        super.startScroll(startX, startY, dx, dy, zero ? 0 : scrollDuration);
    }

    @Override
    public void startScroll(int startX, int startY, int dx, int dy) {
        super.startScroll(startX, startY, dx, dy, zero ? 0 : scrollDuration);
    }

    public int getScrollDuration() {
        return scrollDuration;
    }

    public void setScrollDuration(int scrollDuration) {
        this.scrollDuration = scrollDuration;
    }

    public boolean isZero() {
        return zero;
    }

    public void setZero(boolean zero) {
        this.zero = zero;
    }
}

File: library/banner/src/main/java/com/example/banner/view/ViewPagerScroller.kt
package com.example.banner.view

import android.content.Context
import android.widget.Scroller

class ViewPagerScroller(context: Context?) : Scroller(context) {

    // 滑动速度,值越大滑动越慢，滑动太快会使3d效果不明显
    var scrollDuration = 800
    var isZero = false

    override fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int, duration: Int) {
        super.startScroll(startX, startY, dx, dy, if (isZero) 0 else scrollDuration)
    }

    override fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int) {
        super.startScroll(startX, startY, dx, dy, if (isZero) 0 else scrollDuration)
    }
}